backend/src/git.rs

```diff
@@ -363,6 +560,31 @@ impl Interface {
         Ok(repo.reference_to_annotated_commit(&fetch_head)?)
     }
 
+    /// A code level re-implementation of `git fetch` for a specified branch.
+    /// `git fetch` will sync your local `origin/[BRANCH]` with the remote, but it won't
+    /// merge those changes into your local branch.
+    /// 
+    /// This implementation fetches only the specified branch.
+    /// 
+    /// Returns a reference to the latest commit fetched from remote (`FETCH_HEAD`). This is done if you'd like to merge the remote changes into a local branch.
+    fn git_fetch_branch<'a>(repo: &'a Repository, branch: &'a str) -> Result<AnnotatedCommit<'a>> {
+        let mut remote = repo.find_remote("origin")?;
+    
+        let mut fetch_options = FetchOptions::new();
+        fetch_options.download_tags(git2::AutotagOption::All);
+    
+        // Fetch only the specified branch
+        remote.fetch(
+            &[&format!("refs/heads/{branch}:refs/remotes/origin/{branch}")],
+            Some(&mut fetch_options),
+            None,
+        )?;
+        drop(remote);
+    
+        let fetch_head = repo.find_reference("FETCH_HEAD")?;
+        Ok(repo.reference_to_annotated_commit(&fetch_head)?)
+    }
+
```

Comment:
These two functions should be combined, either always specifying the branch or fetching every branch via wildcard refspecs.